Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,219 @@
+            return AZ::Failure(errorMessage);
+        }
+        QString logFilePath = logFilePathQuery.GetValue();
+        QFile logFile(logFilePath);
+        if (!logFile.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+        {
+            QString errorMessage = QString("%1: %2").arg(tr(ErrorMessages::LogOpenFailureMsg)).arg(logFilePath);
+            QStringToAZTracePrint(errorMessage);
+            return AZ::Failure(errorMessage);
+        }
+
+        EngineInfo engineInfo;
+
+        AZ::Outcome<EngineInfo> engineInfoResult = PythonBindingsInterface::Get()->GetEngineInfo();
+        if (engineInfoResult.IsSuccess())
+        {
+            engineInfo = engineInfoResult.GetValue();
+        }
+        else
+        {
+            QString error = tr("Failed to get engine info.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        QTextStream logStream(&logFile);
+
+        UpdateProgress(tr("Setting Up Environment"));
+
+        auto currentEnvironmentRequest = ProjectUtils::SetupCommandLineProcessEnvironment();
+        if (!currentEnvironmentRequest.IsSuccess())
+        {
+            QStringToAZTracePrint(currentEnvironmentRequest.GetError());
+            return currentEnvironmentRequest;
+        }
+
+        
+
+        m_exportProjectProcess = new QProcess(this);
+        m_exportProjectProcess->setProcessChannelMode(QProcess::MergedChannels);
+        m_exportProjectProcess->setWorkingDirectory(m_projectInfo.m_path);
+
+        
+        QStringList exportArgs = {"export-project", "--export-script", m_projectInfo.m_currentExportScript,
+                                                    "--project-path", m_projectInfo.m_path};
+        QString commandProgram = QDir(engineInfo.m_path).filePath(GetO3DECLIString());
+
+        m_exportProjectProcess->start(commandProgram, exportArgs);
+        if (!m_exportProjectProcess->waitForStarted())
+        {
+            QString error = tr("Exporting project failed to start.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        while (m_exportProjectProcess->waitForReadyRead(MaxExportTimeMSecs))
+        {
+            QString exportOutput = m_exportProjectProcess->readAllStandardOutput();
+
+            logStream << exportOutput;
+            logStream.flush();
+
+            // Show last line of output
+            if (QStringList strs = exportOutput.split('\n', Qt::SkipEmptyParts);
+                !strs.empty())
+            {
+                UpdateProgress(strs.last());
+            }
+
+            if (QThread::currentThread()->isInterruptionRequested())
+            {
+                // QProcess is unable to kill its child processes so we need to ask the operating system to do that for us
+                auto killProcessArgumentsResult = ConstructKillProcessCommandArguments(QString::number(m_exportProjectProcess->processId()));
```

Comment:
I think we have a standard to prefer _auto_ when the type is easily known for example, when for-looping using iterators or `auto something = create<type>(thing)`
```suggestion
                AZ::Outcome<QStringList, QString> killProcessArgumentsResult = ConstructKillProcessCommandArguments(QString::number(m_exportProjectProcess->processId()));
```